OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,286 @@
+    void appendRow(const ETX& indRow, const RowVector& depRow) {
+        validateAppendRow(indRow, depRow);
+
+        _indData.push_back(indRow);
+
+        if(_depData.nrow() == 0 || _depData.ncol() == 0) {
+            try {
+                auto& labels = 
+                    _dependentsMetaData.getValueArrayForKey("labels");
+                if(depRow.size() != labels.size())
+                    throw Exception{"Number of columns in the input row does "
+                            "not match the number of labels."};
+            } catch(std::out_of_range&) {
+                // No operation.
+            }
+            _depData.resize(1, depRow.size());
+        } else
+            _depData.resizeKeep(_depData.nrow() + 1, _depData.ncol());
+            
+        _depData.updRow(_depData.nrow() - 1) = depRow;
+    }
+
+    /** Get row at index.                                                     */
+    RowVector getRowAtIndex(size_t index) const {
+        return _depData.row(index);
+    }
+
+    /** Get row corresponding to the given entry in the independent column.   */
+    RowVector getRow(const ETX& ind) const {
+        auto iter = std::find(_indData.cbegin(), _indData.cend(), ind);
+
+        if(iter == _indData.cend())
+            throw Exception{"Independent column has no entry with value " +
+                    std::to_string(ind)};
+
+        return _depData.row(std::distance(_indData.cbegin(), iter));
+    }
+
+    /** Update row at index.                                                  */
+    RowVector updRowAtIndex(size_t index) {
+        return _depData.updRow(index);
+    }
+
+    /** Update row corresponding to the given entry in the independent column.*/
+    RowVector updRow(const ETX& ind) {
+        auto iter = std::find(_indData.cbegin(), _indData.cend(), ind);
+
+        if(iter == _indData.cend())
+            throw Exception{"Independent column has no entry with value " +
+                    std::to_string(ind)};
+
+        return _depData.updRow(std::distance(_indData.cbegin(), iter));
+    }
+
+    /** Get independent column.                                               */
+    const std::vector<ETX>& getIndependentColumn() const {
+        return _indData;
+    }
+
+    /** Get dependent column.                                                 */
+    ColumnVector getDependentColumnAtIndex(size_t index) const {
+        return _depData.col(index);
+    }
+
+protected:
+    size_t computeNumRows() const override {
+        return _depData.nrow();
+    }
+
+    size_t computeNumColumns() const override {
+        return _depData.ncol();
+    }
+
+    void validateIndependentMetaData() const override {
+        try {
+            _independentMetaData.getValueForKey("labels");
+        } catch(std::out_of_range&) {
+            throw Exception{"Independent metadata does not contain 'labels'."};
+        }
+    }
+
+    void validateDependentsMetaData() const override {
+        unsigned num_cols{};
```

Comment:
> This is a local variable, not in the interface. I think it is okay to let programmers some freedom in choosing how they name their local variables.

I think it would be better for all of us to follow the project naming conventions so that it is easier to read each others' code, and to set a good example for outside contributors. I do agree it doesn't matter much here though so I will defer to other reviewers about how strictly this should be followed. Opinions?